### DISCUSSION TITLE
strongswan scepclient error parsing distinguished name

### DISCUSSION THREAD

**Discussion by thrdgeek** [strongswan/strongswan#1827] [Q&A]
I have Ubuntu Jammy with the following packages installed: libc6, debconf, build-essential, libgmp-dev, libunbound-dev, libldns-dev, libldns-dev, libcurl4-openssl-dev, libcurl4-openssl-dev, network-manager-strongswan, libunbound-dev, pwgen, libstrongswan, network-manager-strongswan, strongswan-charon, strongswan-nm, libstrongswan-extra-plugins, libstrongswan-standard-plugins, strongswan-pki, strongswan-scepclient, strongswan-starter

I am attempting to enroll a client certificate with the following string:
ipsec scepclient --url http://${SCEPSERVER}/CertSrv/mscep/mscep.dll \
		-k 2048 -f -p ${CHALLENGE} \
		--dn "C=${COUNTRY}, S=${STATE}, L=${LOCALITY}, O=${ORGANIZATION}, OU=${ORGANIZATIONALUNIT}, CN=${HOSTNAME}" \
		--subjectAltName dns=${HOSTNAME} \
		--in pkcs1=/etc/ipsec.d/private/privatekey.pem \
		--out cert=/etc/ipsec.d/certs/$HOSTNAME.der \
		--in cacert-enc=caCert-ra-1.der \
		--in cacert-sig=caCert-ra-2.der \
		--debug 4

The error returned is:
error: parsing of distinguished name

I use this same solution with Ubuntu Xenial, Bionic & Focal.  I suspect there is a package missing but I can't seem to identify it.

**Comment by tobiasbrunner** [comment]
> I suspect there is a package missing but I can't seem to identify it.

No missing package could cause that error. It's more likely your env variables are not set and the passed DN is invalid.

**Reply by thrdgeek** [reply]
Thank you.  That was my inital thought but this is the same snippet of a script that does work on Focal.

**Reply by thrdgeek** [reply]
I replaced the variables with actual values, same result.

**Reply by strongX509** [reply]
Are there any additional comma or apostrophe characters contained in the env variables?
